### ISSUE TITLE
clang++ warning -Wundefined-bool-conversion

### ISSUE THREAD

**Issue by mslehto** [sems-server/sems#8]
New warning found with clang++ version greater than 3.4

OS: FreeBSD 10.1
compiler: clang++ 3.6.1 (3.4.1 does not give same warning)
```
$ clang++36 -MMD -Wall -Wno-reorder -Wno-unused -fPIC -g   -O2  -D_DEBUG -D_THREAD_SAFE -D_REENTRANT -DSEMS_VERSION='"1.6-dev-1270-gf634b1d-dirty"' -DARCH='"x86"' -DOS='"freebsd"' -DSYSTEM_SAMPLECLOCK_RATE=32000LL -DSEMS_APP_NAME='"sems"' -DUSE_MONITORING -I/usr/local/include -I ../../core  -c DSMCall.cpp -o DSMCall.o
DSMCall.cpp:112:35: warning: reference cannot be bound to dereferenced null pointer in well-defined C++ code; pointer may be assumed to always convert to true
      [-Wundefined-bool-conversion]
  avar[DSM_AVAR_REQUEST] = AmArg(&req);
                           ~~~~~  ^~~
1 warning generated.
$ clang++36 -v
clang version 3.6.1 (tags/RELEASE_361/final)
Target: i386-portbld-freebsd10.1
Thread model: posix
```

**Comment by mslehto** [comment]
Fixed by commit 0a061366a877f09b6936593eb2cfdb720cba9746

**Comment by mslehto** [comment]
still fixed :)
